### PR TITLE
Add a conflict for pagerfanta 3.0 before allowing it in #143

### PR DESCRIPTION
**Pull request by loic425**
On Sylius\Bundle\GridBundle\Docctrine\DBAL\DataSource, there is a usage of Pagerfanta\Adapter\DoctrineDbalAdapter which does not exist on pagerfanta 3.0.

It will be replaced on #143 
https://github.com/Sylius/SyliusGridBundle/blob/5da00cdd4916473025de064822332ed5144c959f/src/Bundle/Doctrine/DBAL/DataSource.php#L17

### CODE REVIEW

**Review by vvasiloi — APPROVED** [review]

**Review by lchrusciel — COMMENTED** [review]
Lets go with this solution, but we should introduce CONFLICT.md file with some explanation

File: composer.json [hunk 1]
```diff
@@ -46,6 +46,7 @@
         "sylius/grid": "self.version"
     },
     "conflict": {
+        "pagerfanta/pagerfanta": "^3.0",
```
Comment (mbabker) [inline composer.json:49]: You need to conflict the individual packages as well, after the Pagerfanta 2.4 releases there is no guarantee that downstream users are installing the `pagerfanta/pagerfanta` mono-package.

Based on the issue report, `pagerfanta/doctrine-dbal-adapter` needs to be added at a minimum.

You'll also need a conflict blocking versions older than 2.4 as that's when the newer classes were introduced.

Comment (loic425) [reply composer.json:49]: @mbabker Oh you're right.

Comment (loic425) [reply composer.json:49]: @mbabker Do you want to make another PR replacing this one?

**Review by mbabker — APPROVED** [review]

**Comment by loic425** [comment]
@lchrusciel I added the CONFLICTS.md to explain the added conflicts.

**Review by GSadee — APPROVED** [review]

**Comment by lchrusciel** [comment]
Closing in favour of #205

**Pull request closed without merging**